File: FiberEvolutionScraper/src/models/FiberPointDTO.ts
import type BaseModelDTO from "./BaseModelDTO";
import EligibiliteFtthDTO from "./EligibiliteFtthDTO";

export default class FiberPointDTO implements BaseModelDTO
{
    created = new Date();

    lastUpdated = new Date();

    eligibilitesFtth: EligibiliteFtthDTO[] = [];

    signature = "";

    libAdresse = "";

    x = 0.0;

    y = 0.0;

    iconUrl = "";

    opacity = 1;
}

File: FiberEvolutionScraper.Api/Models/FiberPointDTO.cs
using Microsoft.EntityFrameworkCore;
using System.ComponentModel.DataAnnotations.Schema;

namespace FiberEvolutionScraper.Api.Models;

[Table("FiberPoint")]
[PrimaryKey(nameof(Signature))]
public class FiberPointDTO : BaseModelDTO
{
    public string Signature { get; set; }

    public string LibAdresse { get; set; }

    public double X { get; set; }

    public double Y { get; set; }

    [InverseProperty("FiberPoint")]
    public List<EligibiliteFtthDTO> EligibilitesFtth { get; set; } = new List<EligibiliteFtthDTO>();
}